src/sources/aws_s3/sqs.rs

```diff
@@ -198,12 +254,15 @@ impl Ingestor {
                 .clone()
                 .unwrap_or_else(|| "<unknown>".to_owned());
 
-            match self.handle_sqs_message(message, out).await {
+            match self.handle_sqs_message(message).await {
                 Ok(()) => {
                     emit!(SqsMessageProcessingSucceeded {
                         message_id: &message_id
                     });
-                    if self.delete_message {
+                    if self.state.delete_message {
+                        // TODO: SQS supports DeleteMessageBatch, so we could collapse this from 10
+                        // delete calls in serial to a single batch of 10 deletes.  not sure how
+                        // this would interact with E2E acknowledgements, though, in the future.
```

Comment:
AFAIUI deleting SQS messages counts as "acknowledging" the batch of S3 data. As such, batching deletes would require storing sets of the batch status notifiers and then deleting the appropriate set of SQS messages once the right notifiers trigger. This is what I am working on with the Kafka source, with the additional wrinkle that Kafka can only acknowledge events in order.

Reply:
Makes sense, yeah.  Seems like we can leave this as-is for now, given the performance increase is already substantial, and we can wait to see where #7724 lands.